ui: Settings screen according to figma

theborakompanioni opened the pull request:
Adapt `Settings` screen according to Figma.

Differences:
- "Log out" is in section "Wallet" instead of "Settings"

Question:
- Should the "Create new wallet" button be removed? Personally, leaning towards "Yes". (related https://github.com/joinmarket-webui/joinmarket-webui/issues/228)

## 📸 Before
<img src="https://user-images.githubusercontent.com/3358649/170521543-7c7e6278-29d0-4ccb-9013-8756af58749e.png" width=400 />

## 📸 After
<img src="https://user-images.githubusercontent.com/3358649/170521710-54d5b07c-3aa9-4ad7-8ed9-3205dac656aa.png" width=400 />

editwentyone commented:
the Logout is for the whole app, not on wallet level. it should go over Wallet category please :)

theborakompanioni commented:
> maybe even "Log out from Jam"

Hmm.. I am not sure I know what this means.

Currently "logging out" is just forgetting the session key. What should happen, when someone "logs out"?

editwentyone commented:
its should bring you to "login page" https://www.figma.com/file/kfejZJFlwBywvLEnPEmJo1/?node-id=3994%3A73345

but I think we do not have that yet

dnlggr commented:
> the Logout is for the whole app, not on wallet level.

I think these are equivalent things at this time (as @theborakompanioni mentioned) right?

dnlggr commented on src/constants/routes.ts:
```diff
@@ -1,5 +1,6 @@
 export const routes = {
   home: '/',
```
No but I mean `routes.home` and `routes.walletList` are a duplicate, right?

editwentyone commented:
I think that "log out" from Wallet, because its now underneath is misleading. 

logout should be only used when the user logged in somewhere, if that's not the case, we need to rethink what's actually happening.

I would move logout back up again and hide it as long as we don't have login implemented. why would a user logout aka "lock" the wallet and not just close the window?

theborakompanioni commented:
> I would move logout back up again and hide it as long as we don't have login implemented. 

Removed it for now.

> why would a user logout aka "lock" the wallet and not just close the window? 

You are right, the (now removed) "Log out" action, by forgetting the auth token, would have been the same as closing the window.
However, I can relate to some _uneasiness_ when closing a tab with an "unlocked" wallet. A dedicated button would "feel" a little better, albeit there is no difference in what actually happens.

> what happens when we just **close** the window?

The auth token is removed (as `sessionStorage` cleared when the page session ends).

> 
> ~
> 
> so we have:
> 
>     * **create wallet**: creating new wallets, not possible while another wallet is still used and unlocked
> 
>     * **unlock wallet**: wallet is _inactive_, **needs password** to be active
> 
>     * **open wallet**: entering, using, wallet is _active_
> 
>     * **lock wallet**: deactivate wallet, wallet is now _inactive_
> 
>     * **Logout** wallet (misunderstanding): wallet stays active in the background, **needs password** to be opened/ entered again

Maybe something like "Leave wallet", "Exit wallet" or "Close wallet" (although the term "Closing" might be confused by "Locking")?

In any case, I have just removed it for now and will support anything that you think makes most sense from an UI/UX perspective.
Do you think these changes can be merged?

<img src="https://user-images.githubusercontent.com/3358649/172131107-f966c04d-6820-44a3-a431-93580c54f46c.png" width=400 />

dnlggr commented:
Removing seems like n acceptable solution for now. 👍  All the nuances are way to complicated, we really need to figure out a good concept for this.

---

![IMG_F1CC67BC9C25-1](https://user-images.githubusercontent.com/10026790/172151607-4516df3b-1faf-4e68-bfb4-bb5b70fc075b.jpeg)

A little padding for the buttons on mobile would be great.

theborakompanioni commented:
> A little padding for the buttons on mobile would be great.

Yeah, did not change that, as I supposed it was intentional. Any improvements are very welcome! Feel free to just push to this branch!

dnlggr commented:
> Yeah, did not change that

Good point! :D Will push an update.

editwentyone reviewed (approved):
hidden is ok for me, I sketched already something for the next call, and yeah: its confusing :P

0xSaksham commented:
@theborakompanioni, I was taking inspiration for ConfirmModal use case and found out that this feature doesn't work here. If you want I can work on it?

theborakompanioni commented:
> @theborakompanioni, I was taking inspiration for `ConfirmModal` use case and found out that this feature doesn't work here. If you want I can work on it?

I am not quite sure what you are referring to.. can you elaborate?

0xSaksham commented:
Confirm modal wants you to confirm that you really want to do this.

But there's no modal opening up here.

0xSaksham commented:
@theborakompanioni let's discuss this in tomorrow's meeting.

If there's no meeting, we'll discuss this in one-on-one call.
